pkg/api/external/kubernetes/namespace/resource_namespace.go

```diff
@@ -0,0 +1,183 @@
+package namespace
+
+import (
+	"bytes"
+
+	"github.com/pkg/errors"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients/kube/cache"
+	"github.com/solo-io/solo-kit/pkg/api/v1/resources"
+	skkube "github.com/solo-io/solo-kit/pkg/api/v1/resources/common/kubernetes"
+	kubev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubewatch "k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+var _ resources.ResourceNamespaceLister = &kubeResourceNamespaceLister{}
+var _ resources.ResourceNamespaceLister = &kubeClientResourceNamespaceLister{}
+
+// NewKubeClientCacheResourceNamespaceLister will create a new resource namespace lister that requires the kubernestes
+// client and cache.
+func NewKubeClientCacheResourceNamespaceLister(kube kubernetes.Interface, cache cache.KubeCoreCache) resources.ResourceNamespaceLister {
+	return &kubeResourceNamespaceLister{
+		client: NewNamespaceClient(kube, cache),
+	}
+}
+
+// NewKubeClientResourceNamespaceLister will create a new resource namespace lister that requires the kubernetes client
+// interface.
+func NewKubeClientResourceNamespaceLister(kube kubernetes.Interface) resources.ResourceNamespaceLister {
+	return &kubeClientResourceNamespaceLister{
+		kube: kube,
+	}
+}
```

Comment:
I am not familiar with the `KubeCoreCache`, which one is the better approach to getting resources?

Reply:
kube core cache is for shared informers and i think we should generally prefer it for performance